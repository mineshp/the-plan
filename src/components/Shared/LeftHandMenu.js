import React, { Component } from 'react'
import { Icon, Menu } from 'semantic-ui-react'

export default class LeftHandMenu extends Component {
  state = { activeItem: 'view' }

  handleItemClick = (e, { name }) => this.setState({ activeItem: name })

  render() {
    const { activeItem } = this.state

    return (
      <Menu icon='labeled' vertical>
        <Menu.Item name='view' active={activeItem === 'view'} onClick={this.handleItemClick}>
          <Icon name='list layout' />
          View
        </Menu.Item>

        <Menu.Item
            name='create'
            active={activeItem === 'create'}
            onClick={this.handleItemClick}
            href='/project/create'
        >
          <Icon name='add circle' />
          Create
        </Menu.Item>

        <Menu.Item name='search' active={activeItem === 'search'} onClick={this.handleItemClick}>
          <Icon name='search' />
          Search
        </Menu.Item>
      </Menu>
    )
  }
}