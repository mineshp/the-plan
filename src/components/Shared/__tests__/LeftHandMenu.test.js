import React from 'react';
import LeftHandMenu from '../LeftHandMenu';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Menu } from 'semantic-ui-react'

describe('LeftHandMenu Component', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<LeftHandMenu />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('ensure correct menu items found', () => {
        const wrapper = shallow(<LeftHandMenu />);
        expect(wrapper.find(Menu).length).toBe(1);
        expect(wrapper.find(Menu.Item).length).toBe(3);
    });

    it('ensure activeItem is set when a menu item is selected', () => {
        const mockHandleClick = jest.fn();

        const wrapper = shallow(<LeftHandMenu />);
        const firstMenuItem = wrapper.find(Menu.Item).first();
        firstMenuItem.props().onClick(null, { name: 'testabc' });

        expect(wrapper.state().activeItem).toBe('testabc');
    });
});