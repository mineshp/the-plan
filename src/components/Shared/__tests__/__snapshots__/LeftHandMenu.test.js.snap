// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftHandMenu Component renders correctly 1`] = `
<div
  className="ui vertical labeled icon menu"
>
  <a
    className="active item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="list layout icon"
    />
    View
  </a>
  <a
    className="item"
    href="/project/create"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="add circle icon"
    />
    Create
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="search icon"
    />
    Search
  </a>
</div>
`;
